src/date_type.h

```diff
@@ -88,6 +88,13 @@ static const Year DEF_START_YEAR = 1950;
 /** The default scoring end year */
 static const Year DEF_END_YEAR = ORIGINAL_END_YEAR - 1;
 
+/** The default technology progress speed. */
+static const uint8 DEF_TECHNOLOGY_PROGRESS_SPEED = 1;
+/** Frozen technology progress speed. */
+static const uint8 FROZEN_TECHNOLOGY_PROGRESS_SPEED = 0;
+/** The slowest non-frozen technology progress speed, 1/255 times normal. */
+static const uint8 SLOWEST_TECHNOLOGY_PROGRESS_SPEED = 255;
```

Comment:
Agree on this, at least partially. The trouble with a percentage is that you need to make it of the regular speed, as well as be integer, meaning that 1% (one-hundredth) will be the slowest technology speed possible then.
My idea for an approach (without having read the code yet) would be to implement it as ticks-per-technology-day internally, but have the setting in the UI select via some other representation. Honestly, a fraction is the most sensible for large slowdowns.

Reply:
Isn't it already a fraction? Afaict setting 2 means 2x slowdow, setting 10 means 10x. 255 is just some arbitrary limit because setting and counter use byte, can be easily changed to anything else.

Reply:
Let me rephrase then: I agree that a fraction isn't great UI, but it's better than a percentage-based UI which limits the possible range usable. The main downside of a straight 1/x fraction is that you won't be able to set values such as 67% or 40% speed.

Reply:
Not necessarily defending this being set as a fraction, but my prediction is that most people will choose 1/1 (normal and default), 0 (frozen), and 1/2, 1/4, perhaps 1/8...at some point it's so slow that the player probably wants "frozen". But rather than choosing an arbitrary number, I allowed 1/255 as the slowest speed the uint8 will hold.

I can't see a usecase for people wanting to speed up technology progression. It's already quite fast. But if we move to a percentage-based selection I suppose it would be easier to add.

Reply:
For faster progression, I can see the case. Many goal games are pretty fast, just 1-3 hours and while normal progression is mostly ok for these it's not that inconceivable to want it even faster. For example, in the CityMania Christmas event, I made it introduce a new engine pretty much every year. Though in hindsight that may have been a bit too fast it wasn't bad either.